pkg/cloudprovider/provider/baremetal/plugins/tinkerbell/client/common.go

```diff
@@ -28,4 +28,5 @@ var ErrNotFound = errors.New("resource not found")
 const (
 	sqlErrorString    = "rpc error: code = Unknown desc = sql: no rows in result set"
 	sqlErrorStringAlt = "rpc error: code = Unknown desc = SELECT: sql: no rows in result set"
+	sqlErrorNotFound  = "rpc error: code = NotFound desc = not found"
```

Comment:
maybe worth to add a comment why you decided to check errors in that way, as you described here https://github.com/kubermatic/machine-controller/pull/930#discussion_r604979119